nmigen/vendor/quicklogic.py

```diff
@@ -13,28 +13,33 @@ class QuicklogicPlatform(TemplatedPlatform):
     Symbiflow toolchain
     -------------------
     Required tools:
-        * ``synth``
-        * ``pack``
-        * ``place``
-        * ``route``
-        * ``write_fasm``
-        * ``write_bitstream``
+        * ``symbiflow_synth``
+        * ``symbiflow_pack``
+        * ``symbiflow_place``
+        * ``symbiflow_route``
+        * ``symbiflow_write_fasm``
+        * ``symbiflow_write_bitstream``
     The environment is populated by running the script specified in the environment variable
-    ``NMIGEN_ENV_Quicklogic``, if present.
+    ``NMIGEN_ENV_QLSymbiflow``, if present.
     Available overrides:
         * ``add_constraints``: inserts commands in XDC file.
     """
 
     device  = abstractproperty()
     part    = abstractproperty()
 
+    # Since the QuickLogic version of SymbiFlow toolchain is not upstreamed yet
+    # we should distinguish the QuickLogic version from mainline one.
+    # QuickLogic toolchain: https://github.com/QuickLogic-Corp/quicklogic-fpga-toolchain/releases
+    toolchain = "QLSymbiflow"
```

Comment:
The purpose of this attribute is to distinguish multiple different toolchains, such as a proprietary and an open-source one. If there's only a single toolchain it's not necessary.

Reply:
Other than this the PR looks good.

Reply:
I tried to run the quickfeather example without `toolchain` variable but I've got following error:
```jkow@jkow-pc ~> python -m nmigen_boards.quickfeather
Traceback (most recent call last):
  File "/home/jkow/tools/symbiflow-ql/conda/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/jkow/tools/symbiflow-ql/conda/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/jkow/tools/symbiflow-ql/conda/lib/python3.7/site-packages/nmigen_boards-0.1.dev154+g83fe77e-py3.7.egg/nmigen_boards/quickfeather.py", line 91, in <module>
TypeError: Can't instantiate abstract class QuickfeatherPlatform with abstract methods toolchain
```
If this attribute is used to distinguish toolchains only, is it okay to define it e.g. as `toolchain=None`?

Reply:
Sorry, your PR is actually correct—I misremembered how the attribute was supposed to work because we don't have many single-toolchain targets.